### PR TITLE
PT 51/52 ADDED PUBLISH BOOK AND DELETE BOOK MODULE

### PR DESCRIPTION
**Pull request by MartinHwang**
publish book: title and synopsis can be posted
delete book: book can be deleted

### CODE REVIEW

File: src/util/BookSkeleton.js [hunk 1]
```diff
@@ -0,0 +1,75 @@
+import React, { Fragment } from "react";
```
Comment (Harukaichii) [inline src/util/BookSkeleton.js]: Do you really need this file? Can you not modify ScreamSkeleton to work for both Book and Scream/Post?

File: src/components/book/bookCard.js [hunk 2]
```diff
@@ -45,12 +45,12 @@ const styles = {
   },
 };
```
Comment (Harukaichii) [inline src/components/book/bookCard.js:97]: For this file you need to edit the book title CSS!! The current font is white so it's not visible : (

Comment (Harukaichii) [reply src/components/book/bookCard.js:97]: Also https://proudtale.atlassian.net/browse/PT-51 states: 

"From top to bottom order" 
- bookTitle
- Date last updated
- genre tags for top 3 tags (PT-2)

Genre tags you might want to ask someone about but, current order is:
-Date
-Title

**Review by Harukaichii — CHANGES_REQUESTED** [review]
So in terms of redux, I am very happy with this update!! Well done :) Once again some minor styling things~ 

Also 
image

You might want to make it so that on over flow, the dialogue box stretches to accommodate for the content (no scroll bar) .  current behaviour looks a bit weird

File: src/util/BookSkeleton.js (hunk 1 above)
Comment (MartinHwang) [reply src/util/BookSkeleton.js]: Good idea. Thank you. I changed code. Now we have one 'Skeleton.js' for Book and Scream

File: src/components/book/bookCard.js (hunk 2 above)
Comment (MartinHwang) [reply src/components/book/bookCard.js:97]: > For this file you need to edit the book title CSS!! The current font is white so it's not visible : (

some title will be long so I will slice that time using String slice. How about String.slice(0,10)

Comment (MartinHwang) [reply src/components/book/bookCard.js:97]: > Also https://proudtale.atlassian.net/browse/PT-51 states:
> 
> "From top to bottom order"
> 
>     * bookTitle
> 
>     * Date last updated
> 
>     * genre tags for top 3 tags (PT-2)
> 
> 
> Genre tags you might want to ask someone about but, current order is:
> -Date
> -Title

currently, when any book is posted in content creation page, created date is stored on DB. In addition editBook web api include edited dated.  We need to create Genre Schema.

**Comment by MartinHwang** [comment]
> So in terms of redux, I am very happy with this update!! Well done :) Once again some minor styling things~
> 
> Also
> image
> 
> You might want to make it so that on over flow, the dialogue box stretches to accommodate for the content (no scroll bar) . current behaviour looks a bit weird

on Google Chrome display looks pretty good but on Firefox confirmation dialog to go to writing page shows scroll bar. I will fix it.

File: src/components/book/bookCard.js (hunk 2 above)
Comment (Harukaichii) [reply src/components/book/bookCard.js:97]: I agree! sorry I didnt see this earlier. Yea slicing it at 10 sounds good :)

File: src/components/book/DeleteBook.js [hunk 3]
```diff
@@ -82,8 +81,8 @@ class DeleteScream extends Component {
               </DialogTitle>
               <DialogContent className={classes.dialogContent}>
                 <p>
-                  Are you sure you want to{" "}
-                  <span>delete {this.props.title}</span> ?
+                  Are you sure you want to delete{" "}
```
Comment (Harukaichii) [inline src/components/book/DeleteBook.js]: do you need the {""}?

**Review by Harukaichii — APPROVED** [review]
AWESOME WORK MARTIN! Looks great :) Just 1 minnnooor comment. otherwise feel free to merge

**Comment by Harukaichii** [comment]
BTW about the slicing, you can either do it on this branch or leave it till later!

File: src/components/book/DeleteBook.js (hunk 3 above)
Comment (MartinHwang) [reply src/components/book/DeleteBook.js]: If <span>delete {this.props.title}</span> ? is positioned on same line with  'Are you sure you want to{" "}' , {" "} is not necessary. If each is separated in each line, we need it.

**Comment by MartinHwang** [comment]
> BTW about the slicing, you can either do it on this branch or leave it till later!

Long title can make book card broken so I sliced title  to 25 length and added Tooltip to show whole title when move is hovered on title or image

File: src/components/book/DeleteBook.js (hunk 3 above)
Comment (Harukaichii) [reply src/components/book/DeleteBook.js]: ohhh ok I didn't know! :)

**Comment by Harukaichii** [comment]
@MartinHwang  hate to bother you again but I realized this dialogue box's styling is also a lil weird

image

**Comment by Harukaichii** [comment]
image

**Comment by Harukaichii** [comment]
Also the book title's with the a's cardbox still overflow haha

image

**Comment by Harukaichii** [comment]
@MartinHwang Functionality looks good! May I ask that you add a textfield next to the upload image button that displays the name of the uploaded image? that way the user knows they've done something : )

**Pull request closed without merging**